src/commands/goto.ts

```diff
@@ -50,6 +59,70 @@ function executeGoto(gotoType: number, editor: vscode.TextEditor, count: number,
   }
 }
 
+function executeGotoLine(editor: vscode.TextEditor, count: number, extend: boolean, position: string = 'default') {
+  const positions: { [index: string]: {(x: vscode.Selection): number} } = {
+    'first': function (x: vscode.Selection) {
+      return editor.document.lineAt(x.active.line).firstNonWhitespaceCharacterIndex
+    },
+    'end': function (x: vscode.Selection) {
+      return editor.document.lineAt(x.active.line).range.end.character
+    },
+    'start': function (x: vscode.Selection) {
+      return 0
+    },
+    'default': function (x: vscode.Selection) {
+      return 0
+    }
+  }
+  
+  editor.selections = editor.selections.map(x =>
+    {
+      const npos:vscode.Position = new vscode.Position(x.active.line,  (positions[position] || positions['default'])(x));
```

Comment:
Type-checking `position` above correctly should make the conditional check on `positions[position]` useless.

Reply:
sorry have to do those changes at the week end.

Reply:
No problem, thank you for bearing my excessively specific requirements!